pkg/platform/kube/controller/controller.go

```diff
@@ -31,17 +31,18 @@ import (
 )
 
 type Controller struct {
-	logger                logger.Logger
-	namespace             string
-	restConfig            *rest.Config
-	kubeClientSet         kubernetes.Interface
-	nuclioClientSet       nuclioio_client.Interface
-	functionresClient     functionres.Client
-	imagePullSecrets      string
-	functionOperator      *functionOperator
-	projectOperator       *projectOperator
-	functionEventOperator *functionEventOperator
-	platformConfiguration *platformconfig.Config
+	logger                  logger.Logger
+	namespace               string
+	restConfig              *rest.Config
+	kubeClientSet           kubernetes.Interface
+	nuclioClientSet         nuclioio_client.Interface
+	functionresClient       functionres.Client
+	imagePullSecrets        string
+	functionOperator        *functionOperator
+	projectOperator         *projectOperator
+	functionEventOperator   *functionEventOperator
+	cronJobStalePodsDeleter *cronJobStalePodsDeleter
```

Comment:
Cron trigger -> nuclio trigger kind
Cron job -> k8s resource

so maybe you'll have a `CronJobMonitor` function that will delete stale / update with stuff / idk
one of its usage is to delete cron trigger staling pods

And, on top of that, I would suggest from here and below, treat the idea of function has a cronjob resource (as well as deployment/configmap/ingresses/etc) and of its usage is to act as processor-trigger